resume-buddy/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -79,72 +66,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("text", text);
     commentEntity.setProperty("date", date);
     commentEntity.setProperty("uuid", id.toString());
+    commentEntity.setProperty("matchID", getMatchID(email));
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/resume-review.html");
   }
 
-  /** Checks if the id collides with other ids in datastore */
-  private boolean collides(UUID id) {
-    Query query = new Query("Review-comments");
-
-    Filter uuidPropertyFilter = new FilterPredicate("uuid", FilterOperator.EQUAL, id.toString());
-    query.setFilter(uuidPropertyFilter);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-
-    return results.countEntities(FetchOptions.Builder.withDefaults()) != 0;
-  }
-
-  /* Checks if the given user of the given userType has a match of the opposite userType. */
-  public static boolean hasMatch(UserType userType, String email) {
+  public static String getMatchID(String email) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Match");
+    Query query = new Query("User");
     Filter emailFilter;
-    if (userType == UserType.REVIEWEE) {
-      emailFilter = new FilterPredicate("reviewee", FilterOperator.EQUAL, email);
-    } else {
-      emailFilter = new FilterPredicate("reviewer", FilterOperator.EQUAL, email);
-    }
+    emailFilter = new FilterPredicate("email", FilterOperator.EQUAL, email);
     query.setFilter(emailFilter);
     PreparedQuery results = datastore.prepare(query);
-    return results.countEntities(FetchOptions.Builder.withDefaults()) != 0;
+    Entity userEntity = results.asSingleEntity();
+    return (String) userEntity.getProperty("matchID");
   }
 
-  /* If it exists, gets the match of a given user (the user is of type userType). */
-  public static String getMatch(UserType userType, String email) {
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Match");
-    Filter emailFilter;
-    if (userType == UserType.REVIEWEE) {
-      emailFilter = new FilterPredicate("reviewee", FilterOperator.EQUAL, email);
-    } else {
-      emailFilter = new FilterPredicate("reviewer", FilterOperator.EQUAL, email);
-    }
-    query.setFilter(emailFilter);
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      if (userType == UserType.REVIEWEE) {
-        return (String) entity.getProperty("reviewer");
-      } else {
-        return (String) entity.getProperty("reviewee");
-      }
-    }
-    return "";
-  }
-
-  /* add comments from given email and userType(reviewer or reviewee) */
-  public static void addComments(UserType userType, String email, List<Comment> comments) {
+  /* add all comments that have the given match id */
```

Comment:
Nit: I would change this comment to say "Populates the given list of Comments with all comments that have the given match ID."